parlai/agents/image_seq2seq/image_seq2seq.py

```diff
@@ -0,0 +1,349 @@
+            )
+        return obs
+
+    def _dummy_batch(self, batchsize: int, maxlen: int) -> Batch:
+        """
+        Override to include image feats.
+        """
+        return Batch(
+            text_vec=torch.ones(batchsize, maxlen).long().cuda(),
+            label_vec=torch.ones(batchsize, 2).long().cuda(),
+            image=torch.ones(batchsize, self.opt.get('image_features_dim')).cuda(),
+            personalities=torch.ones(batchsize, self.opt.get('embedding_size')).cuda(),
+        )
+
+    def batchify(self, obs_batch: List[Message], sort: bool = False) -> Batch:
+        """
+        Override to handle images.
+        """
+        batch = super().batchify(obs_batch, sort)
+
+        def _process_img(img):
+            if img is not None and isinstance(img, torch.Tensor):
+                if img.dim() == 4:
+                    img = img[0, :, 0, 0]
+                if self.use_cuda:
+                    img = img.cuda()
+                if self.opt.get('fp16'):
+                    img = img.half()
+                else:
+                    img = img.float()
+
+            return img
+
+        if type(batch.image) == list and any(b is not None for b in batch):
+            images = []
+            for img in batch.image:
+                images.append(_process_img(img))
+            batch.image = images
+        return batch
+
+    def _model_input(self, batch: Batch) -> Tuple[torch.Tensor, List[object]]:
+        return (batch.text_vec, batch.image)
+
+    def load_state_dict(self, state_dict: Dict[str, torch.Tensor]):
+        """
+        Override for custom loading.
+
+        Three reasons:
+            1. When using an init model without an image encoder
+            2. When using an init model with only an encoder provided
+                In this case, we may need to add the START token to the state_dict
+            3. When using an init model without image tokens in the embeddings
+        """
+        # Case 1 -> No Image Encoder
+        if 'encoder.image_encoder.0.weight' not in state_dict:
+            for k, v in self.model.encoder.image_encoder.state_dict().items():
+                state_dict[f'encoder.image_encoder.{k}'] = v
+
+        # Case 2 -> Only an Encoder provided
+        if not (any('decoder' in state_key for state_key in state_dict)):
+            for k, v in self.model.decoder.state_dict().items():
+                state_dict[f'decoder.{k}'] = v
+            state_dict['decoder.embeddings.weight'] = state_dict['embeddings.weight']
+            if 'START' not in state_dict:
+                state_dict['START'] = self.model.START
+
+        cnt = 0
+        if self.opt['init_model'] is not None:
+            try:
+                self.model.load_state_dict(state_dict)
+                return
+            except Exception as e:
+                # Case 3 --> Check for Embedding Diffs
+                print('exception in loading state dict: {}'.format(e))
```

Comment:
interesting... do we always want to do this? or do we want to surface the error?

Reply:
for now I assume that the error is embedding based and print the exception; if there is some other error that happens, it'll raise later as I still call `self.model.load_state_dict` after this try/catch

Reply:
Can you leave a comment explaining this? My instinct was definitely like "wow, are we sure we want to catch all exceptions here?" It might be worth trying to narrow down beyond just "Exception" to what kind of exception we're actually expecting (KeyError?)

Reply:
i will change the exception here to make it more clear what we're handling